arithmetization/src/main/java/net/consensys/linea/sequencer/txselection/selectors/MaxBlockSizeTransactionSelector.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package net.consensys.linea.sequencer.txselection.selectors;
+
+import static net.consensys.linea.sequencer.txselection.LineaTransactionSelectionResult.BLOCK_BYTES_SIZE_OVERFLOW;
+
+import java.util.List;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.hyperledger.besu.datatypes.PendingTransaction;
+import org.hyperledger.besu.datatypes.Transaction;
+import org.hyperledger.besu.evm.log.Log;
+import org.hyperledger.besu.plugin.data.TransactionProcessingResult;
+import org.hyperledger.besu.plugin.data.TransactionSelectionResult;
+import org.hyperledger.besu.plugin.services.txselection.PluginTransactionSelector;
+
+@Getter
+@Slf4j
+public class MaxBlockSizeTransactionSelector implements PluginTransactionSelector {
+  private int cumulativeBlockBytesSize = EMPTY_L1_BLOCK_SIZE;
+  private final int maxBytesPerBlock;
+  public static final int AVERAGE_TX_RLP_SIZE = 400;
+  public static final int FROM_ADDRESS_SIZE = 20;
+  public static final int EMPTY_L1_BLOCK_SIZE = 64 + 63 + 64 + 64;
+  public static final int LOG_SIZE = 32;
+
+  public MaxBlockSizeTransactionSelector(int maxBytesPerBlock) {
+    this.maxBytesPerBlock = maxBytesPerBlock;
+  }
+
+  @Override
+  public TransactionSelectionResult evaluateTransactionPreProcessing(
+      PendingTransaction pendingTransaction) {
+    return TransactionSelectionResult.SELECTED;
+  }
+
+  @Override
+  public TransactionSelectionResult evaluateTransactionPostProcessing(
+      final PendingTransaction pendingTransaction,
+      final TransactionProcessingResult processingResult) {
+
+    int L1TransactionSize =
+        calculateL1TransactionSize(pendingTransaction, processingResult.getLogs());
+    if (isTransactionExceedingL1BlockSizeLimit(L1TransactionSize)) {
+      log.atTrace()
+          .setMessage(
+              "Cumulative block size in bytes including tx {} is {} greater than the max allowed {}, skipping tx")
```

Comment:
suggestion -
              "tx {hash} size {} would cause block (cumulative size {}) to exceed max {} bytes, skipping tx"
eg
tx 0x1234 (size 52) would cause block (cumulative size 499) to exceed max 500 bytes, skipping tx

Reply:
done